src/exo/new_eff.py

```diff
@@ -1268,6 +1268,11 @@ def add_name(name):
 
 class ContextExtraction:
     def __init__(self, proc, stmts):
+        if len(stmts) == 0:
+            proc = proc.update(
+                body=[LoopIR.Pass(eff=None, srcinfo=proc.srcinfo)] + proc.body
+            )
+            stmts = [proc.body[0]]
```

Comment:
Is there a better way to do this?

Reply:
Remove this change and call `Check_ExprBound` with dummy statement?